pkg/util/kubernetes/hpa/hpa_test.go

```diff
@@ -32,121 +32,264 @@ func (d *fakeDatadogClient) QueryMetrics(from, to int64, query string) ([]datado
+			for _, m := range externalMetrics {
+				m.Timestamp = 0
+				strippedTs = append(strippedTs, m)
+			}
+
+			assert.ElementsMatch(t, tt.expected, strippedTs)
+		})
+	}
+}
+
+func TestHPAProcessor_ComputeDeleteExternalMetrics(t *testing.T) {
+	tests := []struct {
+		desc     string
+		list     []*autoscalingv2.HorizontalPodAutoscaler
+		emList   []custommetrics.ExternalMetricValue
+		expected []custommetrics.ExternalMetricValue
+	}{
 		{
-			caseName: "no hpa for metric",
-			metrics: []custommetrics.ExternalMetricValue{
+			"Delete invalid metric",
+			[]*autoscalingv2.HorizontalPodAutoscaler{
 				{
-					MetricName: "requests_per_s",
-					Labels:     map[string]string{"bar": "baz"},
-					HPA:        custommetrics.ObjectReference{Name: "foo", Namespace: "default", UID: "1111"},
-					Timestamp:  12,
-					Value:      1,
+					ObjectMeta: v1.ObjectMeta{
+						UID: types.UID(5),
+					},
+				},
+				{
+					ObjectMeta: v1.ObjectMeta{
+						UID: types.UID(7),
+					},
+				},
+			},
+			[]custommetrics.ExternalMetricValue{
+				{
+					MetricName: "requests_per_s_one",
+					Labels:     map[string]string{"foo": "bar"},
+					Valid:      true,
+					HPA: custommetrics.ObjectReference{
+						UID: string(5),
+					},
+				},
+				{
+					MetricName: "requests_per_s_two",
+					Labels:     map[string]string{"foo": "bar"},
 					Valid:      false,
+					HPA: custommetrics.ObjectReference{
+						UID: string(6),
+					},
+				},
+				{
+					MetricName: "requests_per_s_three",
+					Labels:     map[string]string{"foo": "bar"},
+					Valid:      false,
+					HPA: custommetrics.ObjectReference{
+						UID: string(7),
+					},
+				},
+			},
+			[]custommetrics.ExternalMetricValue{
+				{
+					MetricName: "requests_per_s_two",
+					Labels:     map[string]string{"foo": "bar"},
+					Valid:      false,
+					HPA: custommetrics.ObjectReference{
+						UID: string(6),
+					},
 				},
 			},
-			expected: []custommetrics.ExternalMetricValue{},
 		},
 	}
 
-	for i, testCase := range testCases {
-		t.Run(fmt.Sprintf("#%d %s", i, testCase.caseName), func(t *testing.T) {
-			store, client := newFakeConfigMapStore(t, "default", fmt.Sprintf("test-%d", i), testCase.metrics)
-			hpaCl := &HPAWatcherClient{clientSet: client, store: store}
-
-			if testCase.hpa != nil {
-				_, err := client.
-					AutoscalingV2beta1().
-					HorizontalPodAutoscalers("default").
-					Create(testCase.hpa)
-				require.NoError(t, err)
-			}
-
-			hpaCl.gc() // force gc to run
-
-			allMetrics, err := store.ListAllExternalMetricValues()
-			require.NoError(t, err)
-			assert.ElementsMatch(t, testCase.expected, allMetrics)
+	for i, tt := range tests {
+		t.Run(fmt.Sprintf("#%d %s", i, tt.desc), func(t *testing.T) {
+			externalMetrics := ComputeDeleteExternalMetrics(tt.list, tt.emList)
+			assert.ElementsMatch(t, tt.expected, externalMetrics)
 		})
 	}
 }
 
-func TestHPAWatcherUpdateExternalMetrics(t *testing.T) {
+func TestHPAProcessor_ProcessHPAs(t *testing.T) {
```

Comment:
Same thing here `TestHPAProcessor_ProcessHPAs` should be `TestProcessor_ProcessHPAs`.